pkg/monitor/clustermonitor.go

```diff
@@ -210,6 +224,10 @@ func (m *Monitor) updateCluster(managedCluster *clusterv1.ManagedCluster) {
 	if found && clusterID != m.ManagedClusterInfo[clusterIdx].ClusterID {
 		// If the cluster ID has changed update it - otherwise do nothing.
 		glog.Infof("Updating %s from Insights cluster list", clusterToUpdate)
+		lock.Lock()
+		defer lock.Unlock()
+		m.ClusterNeedsCCX[clusterID] = m.ClusterNeedsCCX[m.ManagedClusterInfo[clusterIdx].ClusterID]
+		delete(m.ClusterNeedsCCX, m.ManagedClusterInfo[clusterIdx].ClusterID)
```

Comment:
Can we  lookup if the clusterId is present in the list - if it is not present , can lead to runtime errors.